### PR TITLE
Fix various typos, repeated words, align some spelling to LDP.

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
partially revamped from #16712
- fall thru -> fall through
- time stamp -> timestamp
- file name -> filename
- host name -> hostname

Only fix a few `low level =>low-level` to reduce the PR extent.

Fix a few pointer compare test nits, while fixing some typos.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: ACKNOWLEDGEMENTS.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Acknowlegements
+Acknowledgements
 ===============
```
Comment (t8m) [inline ACKNOWLEDGEMENTS.md:2]: You need to fix the underlining as well.

**Review by t8m — COMMENTED** [review]
IMO this is for master branch only. The fixes of the documentation would be probably eligible for 3.0 backport, however the comments changes are not something we would backport.

File: ACKNOWLEDGEMENTS.md (hunk 1 above)
Comment (FdaSilvaYY) [reply ACKNOWLEDGEMENTS.md]: Fixed !

**Comment by FdaSilvaYY** [comment]
I may submit a PR to backport documentation fixes only, if you want.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
> I may submit a PR to backport documentation fixes only, if you want.

Feel free to submit it.

**Comment by FdaSilvaYY** [comment]
Ping @openssl/commiters, needs a 2nd reviewer.

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the contributions.

**Pull request closed without merging**